### PR TITLE
Implement basic UI for sharing libraries.

### PR DESCRIPTION
**Pull request by yipingliao**
This is not according to the design wireframes yet, because they are still not finalized. However, I wanted to get the basic functionality coded up; this will allow us (people in the library experiment) to actually invite people through the website.

@andrewconner cc @2is10 
https://app.asana.com/0/15523651812135/15523601193412

### CODE REVIEW

**Comment by andrewconner** [comment]
Wahoo!

File: kifi-backend/modules/shoebox/angular/src/libraries/library.js [hunk 1]
```diff
@@ -134,4 +136,157 @@ angular.module('kifi')
...
+          // For now, we are only supporting inviting one person at a time.
+          var invitees = [
+            {
+              type: !!result.id ? 'user' : 'email',
+              id: !!result.id ? result.id : result.email,
+              access: 'read_only'  // Right now, we're only supporting read-only access.
+            }
+          ];
+
+          // console.log('Library ID: ' + scope.library.id + ' Invitee: ' + JSON.stringify(invitees));
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/libraries/library.js:275]: @ahsu1230 and I found yesterday that the endpoint is a little flaky (sometimes it returns 500, sometimes it processes the invite successfully). Leaving the log messages in for debugging for a while longer.

File: kifi-backend/modules/shoebox/angular/src/libraries/library.styl [hunk 2]
```diff
@@ -0,0 +1,7 @@
+.kf-library-share
+  .social-invite-element
+    &:hover
+      cursor: pointer
+
+    &.selected
+      background-color: #eee
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/libraries/library.styl]: Oops, will add newline.

File: kifi-backend/modules/shoebox/angular/src/libraries/library.js (hunk 1 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/libraries/library.js:275]: Interesting. Can you trigger it intentionally?

File: kifi-backend/modules/shoebox/angular/src/libraries/libraryShareSearch.tpl.html [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<div class="social-invite-search">
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/libraries/libraryShareSearch.tpl.html:1]: Some CSS classes are borrowed from the friend search dropdown right now. There will be a lot of CSS changes later when we implement the design, so I have not cleaned this up yet.

File: kifi-backend/modules/shoebox/angular/src/libraries/library.js (hunk 1 above)
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/libraries/library.js:275]: Unfortunately, no. :(

File: kifi-backend/modules/shoebox/angular/src/common/services/routeService.js [hunk 4]
```diff
@@ -62,6 +62,9 @@ angular.module('kifi')
       friendRequest: function (id) {
         return env.xhrBase + '/user/' + id + '/friend';
       },
+      contactSearch: function (opt_query) {
+        return route('/user/contacts/search' + (!!opt_query ? '?query=' + opt_query : ''));
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/common/services/routeService.js]: FYI—coercion to boolean is implicit in this context, so `!!` is unnecessary

Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/common/services/routeService.js]: I got a little carried away with `!!` in this pull request. :) I'll go back and remove any that are not necessary.

**Pull request merged**